redash/query_runner/python.py

```diff
@@ -187,9 +184,9 @@ def run_query(self, query):
             #       One option is to use ETA with Celery + timeouts on workers
             #       And replacement of worker process every X requests handled.
 
-            exec(code) in restricted_globals, script_locals
+            exec(code) in restricted_globals, self._script_locals
 
-            json_data = json.dumps(self._result)
```

Comment:
I could merge if it exists already, but doesn't feel like it worth the hassle. People shouldn't abuse this :)

See fix here: https://github.com/EverythingMe/redash/commit/07b88d0b53189cf777c252029d361ad00c05fe9e
(removed the need for WeakRef too)